### DISCUSSION TITLE
Highlight selected Spectrumcard in Dataprovider

### DISCUSSION THREAD

**Discussion by sweet-fl** [Budibase/budibase#13837] [Help] [answered]
Hey! I've been looking for a way to highlight the selected spectrumcard within a dataprovider, but to no avail. I presume it would require some code which is not a strength of mine. Would be grateful if you could help me out. :))

Here's the current structure:

![image](https://github.com/Budibase/budibase/assets/171412927/3ca35586-dfab-463d-92d6-a83d3ba52435)

I have my order previews contained within the spectrumcards and when I select one of the card in production, a new window opens to the left with all the order information displayed. As of now, when I select the relevant order spectrumcard, it doesn't highlight so I know which order I've selected from the list. I'd like it to highlight so it looks something like that:

![tempsnip](https://github.com/Budibase/budibase/assets/171412927/119cda38-1739-4222-b036-091610fcf50e)

Any ideas how I could do this? Thanks a bunch in advance!

**Comment by mikesealey** [comment, marked as answer]
Hi @sweet-fl 

This is possible without too much trouble, just a little bit of CSS I believe.

It looks like you're using a Row Explorer component that's been ejected from the block. If you click on your Spectrumcard in the Component Tree you should see that it has an On Click action setting it's value in State. This is quite useful, because each Spectrumcard On Click will overwrite that State value so that there can only be one at a time.

With your Spectrumcard selected, go to Condtions in the settings panel on the right.
![image](https://github.com/Budibase/budibase/assets/39269380/16a296b8-4f31-4836-8452-667865ad08a6)

You'll want to set two conditions 
(Roughly in plain English)
If the Repeater Employees ID is equal to the State Value that was set when clicked => Show my styling
If the Repeater Employees ID is NOT equal to the State value that was set when clicked => Don't show my styling

In Budibase Settings, they look a little like this

- `Update Setting`
- `Custom CSS`
- `TO (insert your styling here)`
- `IF {{ Repeater.<your_table_name_here>._id`
- `EQUALS`
- `BINDING`
- `{{ State.<your_value_here> }}`

Which will apply the styling to this block, when this block is clicked

And then the following
- `Update Setting`
- `Custom CSS`
- `TO (leave blank to remove styling)`
- `IF {{ Repeater.<your_table_name_here>._id`
- `NOT EQUALS`
- `BINDING`
- `{{ State.<your_value_here> }}`

Which will remove the styling from this component when a different Spectrumcard is clicked.

(In my example I use `border: 3px solid red` but your styling is entirely up to you.
![image](https://github.com/Budibase/budibase/assets/39269380/d9adf0f9-0b8b-4428-8163-110623cd444e)

Here's an export of my app that I used to demonstrate this using the sample data: [13837-export-1717411065570.tar.gz](https://github.com/user-attachments/files/15532607/13837-export-1717411065570.tar.gz).  If you're unsure how to import it, you can read the [Budibase Documentation here](https://docs.budibase.com/docs/export-and-import-apps#import-an-application). Feel free to have a look around.

We also have an active and welcoming community [discord server ](https://discord.com/invite/ZepTmGbtfF) where you might find answers and inspiration from other Budibase users.

I hope that gets you unstuck.

**Reply by sweet-fl** [reply]
Hey, @mikesealey,

Thank you very much for the time taken to help me out with this. I’ll try to see if I’ll manage to update the settings and get back here in case I encounter any trouble!

Have a great evening!
